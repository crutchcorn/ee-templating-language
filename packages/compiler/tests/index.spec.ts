import * as path from "node:path";
import {fileURLToPath} from 'node:url';
import {dirname} from 'node:path';

import {test, expect} from "vitest";

import {compile} from "../src/index.js";
import * as fs from "node:fs";

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

test("compiles a basic file", () => {
  const result = compile(fs.readFileSync(path.resolve(__dirname, "../../../sample/test.json.ee"), "utf8"), {});

  expect(result).toMatchSnapshot();
});

test("compiles an array-interpolated file", () => {
  const result = compile(fs.readFileSync(path.resolve(__dirname, "../../../sample/array.json.ee"), "utf8"), {});

  expect(result).toMatchSnapshot();
});
