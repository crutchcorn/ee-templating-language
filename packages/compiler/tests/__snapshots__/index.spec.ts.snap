// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compiles a basic file 1`] = `
"
{
    "test": 123
}
"
`;

exports[`compiles a top-level await file 1`] = `
"
{
    "test": 123
}
"
`;

exports[`compiles an array-interpolated file 1`] = `
"
{
    "test": 123
}
"
`;
